web/src/components/apps/AppDetailPage.tsx

```diff
@@ -279,94 +281,64 @@ function AppDetailPage(props: Props) {
     }
   };
 
-  /**
-   *  Runs on mount and on update. Also handles redirect logic
-   *  if no apps are found, or the first app is found.
-   */
-  const checkForFirstApp = async () => {
-    const { rootDidInitialAppFetch, appsList } = props;
-    if (!rootDidInitialAppFetch) {
-      return;
-    }
-    state.checkForFirstAppJob?.stop?.();
-    const firstApp = appsList?.find((app) => app.name);
-
-    if (firstApp) {
-      history.replace(`/app/${firstApp.slug}`);
-      getApp(firstApp.slug);
-    } else if (props.isHelmManaged) {
-      history.replace("/install-with-helm");
-    } else {
-      history.replace("/upload-license");
-    }
-  };
-
   // Enforce initial app configuration (if exists)
   useEffect(() => {
-    const { app } = state;
-
-    setCurrentApp(props.appsList?.find((w) => w.slug === params.slug));
-
+    // Handle updating the theme state when switching apps.
+    if (selectedApp?.iconUri) {
+      const { navbarLogo, ...rest } = theme.getThemeState();
+      if (navbarLogo === null || navbarLogo !== selectedApp.iconUri) {
+        theme.setThemeState({
+          ...rest,
+          navbarLogo: selectedApp.iconUri,
+        });
+      }
+    }
     // Refetch app info when switching between apps
-    if (app && !state.loadingApp && params.slug != app.slug) {
-      getApp();
+    if (selectedApp && !appsIsLoading && params.slug !== selectedApp.slug) {
+      refetchApps();
       checkIsVeleroInstalled();
       return;
     }
 
     // Handle updating the theme state when switching apps.
     // Used for a fresh reload
     if (history.location.pathname === "/apps") {
-      checkForFirstApp();
       // updates state but does not cause infinite loop because app navigates away from /apps
       return;
     }
 
-    if (app) {
-      const downstream = app?.downstream;
-      if (downstream?.pendingVersions?.length) {
-        const firstVersion = downstream.pendingVersions.find(
-          (version) => version?.sequence === 0
-        );
-        if (firstVersion?.status === "pending_config") {
-          history.push(`/${app.slug}/config`);
-          return;
-        }
+    // find if any app needs configuration and redirect to its configuration flow
+    const appNeedsConfiguration = appsList?.find((app) => {
+      return app?.downstream?.pendingVersions?.length > 0;
+    });
+    if (appNeedsConfiguration) {
+      const downstream = appNeedsConfiguration.downstream;
+      const firstVersion = downstream.pendingVersions.find(
+        (version: Version) => version?.sequence === 0
+      );
+      if (firstVersion?.status === "pending_config") {
+        history.push(`/${appNeedsConfiguration.slug}/config`);
+        return;
       }
     }
-  }, [state.app]);
+  }, [selectedApp]);
 
   useEffect(() => {
+    refetchApps();
     if (history.location.pathname === "/apps") {
-      state.checkForFirstAppJob.start(checkForFirstApp, 2000);
       return;
     }
-    getApp();
+    // getApp();
```

Comment:
```suggestion
```

Reply:
okay i'll clean this up in another pr. don't want to kick off the ci again